--- a/ovos_core/skill_manager.py
+++ b/ovos_core/skill_manager.py
@@ -162,7 +162,10 @@
 
     def _sync_skill_loading_state(self):
         """Synchronize the loading state of skills with the current system state."""
-        resp = self.bus.wait_for_response(Message("ovos.PHAL.internet_check"))
+        resp = self.bus.wait_for_response(
+            Message("ovos.PHAL.internet_check"),
+            timeout=3
+        )
         network = False
         internet = False
         if not self._gui_event.is_set() and is_gui_connected(self.bus):
